opm/simulators/wells/MultisegmentWellSegments.cpp

```diff
@@ -740,7 +740,8 @@ pressureDropValve(const int seg) const
     const EvalWell friction_pressure_loss =
         mswellhelpers::frictionPressureLoss(additional_length, diameter, area, roughness, density, mass_rate, visc);
 
-    const double area_con = valve.conCrossArea();
+    const ValveUDAEval uda_eval {summary_state, this->well_.name(), static_cast<size_t>(seg)};
```

Comment:
Looking at the [the parameter usage](https://github.com/OPM/opm-common/blob/1594680f357d3c31376d1588701a826a2d3b4468/src/opm/input/eclipse/Schedule/MSW/Valve.cpp#L42-L43), the third constructor parameter appears to be treated as a zero-based segment number.  However, `seg` is not that in this context.  Rather, `seg` is a number in the [range 0..#segments - 1](https://github.com/OPM/opm-simulators/blob/3f436501d2fe213cb9b6b325c420da6c915e2ff1/opm/simulators/wells/MultisegmentWell_impl.hpp#L1622-L1624), in a possibly reordered sequence of segments.  If you really want the segment number here, you can use
```C++
segment_set[seg].segmentNumber()
```
instead.  That will give you the one-based segment number directly.

Reply:
It's the segment number I actually want, thanks!